### PR TITLE
[WIP] Username Checker Modules

### PR DESCRIPTION
**Pull request by prince-7**
Username Checker module as mentioned in https://github.com/saeeddhqan/Maryam/issues/52.
username
Some work is still remaining such as:
- [x] Add More Social media sites.
- [x] Add error message for social media sites.

### CODE REVIEW

File: modules/search/username_checker.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	for results in concurrent.futures.as_completed(futures):
+		print(f"Found {len(output['sites'])} accounts" , end= '\r')
+	print('\n')
+
+
+def check(self,url,site, data):
+	global output
+	try:
+		req = self.request(url)
+		for error in data[site]['error'] :
```
Comment (saeeddhqan) [inline modules/search/username_checker.py]: put  _for_ statement on the else block.

File: data/username_checker.json [hunk 2]
```diff
@@ -0,0 +1,662 @@
+{
```
Comment (saeeddhqan) [inline data/username_checker.json:1]: Compress the JSON file.
Where is the source of this file? take place the source of JSON file in the meta variable.

File: modules/search/username_checker.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import re
```
Comment (saeeddhqan) [inline modules/search/username_checker.py]: You didn't use re.

File: modules/search/username_checker.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	'version': '0.1',
+	'description': 'Search your query across 100+ social networks and show the results.',
+	'options': (
+		('query', None, True, 'Query string', '-q', 'store', str),
+		('thread', 8, False, 'The number of sites that are being checked per round(default=8)', '-t', 'store', int),
+	),
+    'examples': ('username_checker -q <QUERY> --output',)
+}
+
+output = {'sites': {}}
```
Comment (saeeddhqan) [inline modules/search/username_checker.py]: output => OUTPUT
sites => links

File: modules/search/username_checker.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+		return
+
+def module_api(self):
+	query = self.options['query']
+	filepath = os.path.join(os.getcwd(), 'data','username_checker.json')
+	file = open(filepath)
+	data = json.loads(file.read())
+	thread(self, data, query,self.options['thread'])
+
+	self.save_gather(output, 'search/username_checker', query, output=self.options.get('output'))
```
Comment (saeeddhqan) [inline modules/search/username_checker.py]: 'osint/username_search'

**Comment by saeeddhqan** [comment]
It belongs to osint modules with the name of username_search.

File: modules/search/username_checker.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+		output['sites'][site] = url
+	except Exception as e:
+		self.error(f"Not fount on {site}")
+		return
+
+def module_api(self):
+	query = self.options['query']
+	filepath = os.path.join(os.getcwd(), 'data','username_checker.json')
+	file = open(filepath)
+	data = json.loads(file.read())
```
Comment (z3y50n) [inline modules/search/username_checker.py]: Use `with open()`

File: modules/search/username_checker.py [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+	thread(self, data, query,self.options['thread'])
+
+	self.save_gather(output, 'search/username_checker', query, output=self.options.get('output'))
+	return output
+
+def module_run(self):
+	self.alert_results(module_api(self))
```
Comment (z3y50n) [inline modules/search/username_checker.py]: It could be better to use `self.table` to display your results

File: data/username_checker.json (hunk 2 above)
Comment (prince-7) [reply data/username_checker.json:1]: > Where is the source of this file? take place the source of JSON file in the meta variable.

I did not understand this.

Comment (prince-7) [reply data/username_checker.json:1]: Also will compress the JSON file after I have added all the sites.

Comment (saeeddhqan) [reply data/username_checker.json:1]: Did you write the JSON file by yourself?

Comment (prince-7) [reply data/username_checker.json:1]: here is a list of social networks I am referring to https://github.com/sherlock-project/sherlock/blob/master/sites.md

Comment (saeeddhqan) [reply data/username_checker.json:1]: Therefore, write this line in the meta:
`'sources': ('https://github.com/sherlock-project/sherlock',),`

**Comment by z3y50n** [comment]
Everything seems good I will merge. Thank you!

**Review by z3y50n — APPROVED** [review]

**Comment by prince-7** [comment]
> Everything seems good I will merge. Thank you!

Wait json files needs to be updated

**Comment by prince-7** [comment]
I will make a new PR for more sites to json file, no worries

**Pull request merged**